File: TD3/typescript/session3/td/CharAt.ts
import { Visiteur } from "./Visiteur"

export class CharAt implements Visiteur<string> {

    private index : number;
    private compteur : number = 0;
    private charTrouve : string;

    public constructor(index : number) {
        this.index = index;
    }

    public casVide() : string {
        return this.charTrouve;
    }

    public casCons(n : string, r : string) : string {
        if (this.compteur == this.index) {
            this.compteur++;
            this.charTrouve = n;
            return n;
        } else {
            this.compteur++;
            return r;
        }
    }
}

File: Java/src/session3/td/CharAt.java
package session3.td;

public class CharAt implements Visiteur<Character>{

	private int index;
	private int compteur = 0;
	private char charTrouve;
	
	public CharAt(int index) {
		this.index = index;
	}

	
	@Override
	public Character casVide() {
		return charTrouve;
	}

	@Override
	public Character casCons(char str, Character r) {
		
		if (compteur == index) {
			charTrouve = str;
			compteur++;
			return str;
		}
		compteur++;
		return r;
	}
}
